nautobot_floor_plan/forms.py

```diff
@@ -184,8 +346,53 @@ class Meta:
         fields = "__all__"
         exclude = ["allocation_type", "on_group_tile"]  # pylint: disable=modelform-uses-exclude
 
+    def _convert_label_to_position(self, value, axis, fp_obj):
+        """Wrapper for the LabelToPositionConverter."""
+        try:
+            converter = LabelToPositionConverter(value, axis, fp_obj)
+            absolute_position, _ = converter.convert()
+            return absolute_position, None  # Return None for the error when successful
+        except ValueError as e:
+            return None, forms.ValidationError(str(e))
+
+    def _clean_custom_origin(self, field_name, axis):
+        """Clean method for custom label origins."""
+        fp_obj = self.fields["floor_plan"].queryset.get(id=self.data.get("floor_plan"))
```

Comment:
Maybe?

```suggestion
        fp_obj = FloorPlan.objects.get(id=self.data.get("floor_plan"))
```

Though actually, could we just do:

```py
        fp_obj = self.cleaned_data.get("floor_plan")
```

Same comment on similar code below.